scalafmt-tests/src/test/resources/default/Lambda.stat

```diff
@@ -115,12 +115,13 @@ def foo(x: Int) = x match {
     (y: Int) => -y
   }
 >>>
-def foo(x: Int) = x match {
-  case 1 =>
-    (y: Int) => y
-  case _ =>
-    (y: Int) => -y
-}
+def foo(x: Int) =
```

Comment:
Don't understand why `x match {` was not left on its line (like in previous test with single line lambda)

Reply:
this is because `newlines.alwaysBeforeMultilineDef` is `true` by default but was ignored in some cases (including this one). one of the commits in this PR fixes this inconsistency, so people can simply set the flag to `false` if this behaviour is not desired.

Reply:
But why [this](https://github.com/scalameta/scalafmt/pull/1779/files#diff-a0f91ca7765a8af4af56f16fbffd361eR100-R109) test still correct?

Should it be
```scala
def foo(x: Int) =
  x match {
    case 1 => (y: Int) => y
    case _ => (y: Int) => -y
  }
```
?

Reply:
i think it's because that test is marked as `SKIP`

Reply:
:smile: reasonable

Sorry for false alarm

Reply:
nothing to be sorry about; in fact, i really appreciate that you review my changes carefully, the end result is superior, two heads are better than one!